### DISCUSSION TITLE
[BUG] QuestionExec postsave no setea valores en formulario

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#83] [4. Bug Reports] [answered]
Hola 
Me podrían revisar este questions exec plz? :). Me ayudó ignacio en hacerlo, pero creo que lo escribi mal :( porque no funciona cuando se envia el survey :(
```
    "exec" : {
        "postsave" : {
            "src" : "function run() {const peso = Number(context['responses#peso_hepatopancreaticobiliar']);const altura = Number(context['responses#talla_hepatopancreaticobiliar']);var imc = peso / (altura**2);commands.push( {         cmd: 'SET_RESPONSES',   value: imc  });  commands.push(  {  cmd: 'SET_READONLY',    value: 'true' });  return commands;}", 
            "context" : "responses#peso_hepatopancreaticobiliar,responses#talla_hepatopancreaticobiliar"
        }
    }
```

gracias :)

**Comment by acedward** [comment, marked as answer]
Para poder debugear esta implementación

## PASO 1
Agrega console.log(...) para que aparezcan en la consola

Los verás así:
![image](https://user-images.githubusercontent.com/3951895/115032649-88f5ea00-9e97-11eb-8cd8-f7a0f9752a65.png)

**Reply by acedward** [reply]
## PASO 2

Haz click sobre VM9:4 
Verás algo así
![image](https://user-images.githubusercontent.com/3951895/115032896-d3776680-9e97-11eb-9640-bae485497595.png)

y luego sobre pretty-print

![image](https://user-images.githubusercontent.com/3951895/115033021-ef7b0800-9e97-11eb-973a-4b11a233c285.png)

**Reply by acedward** [reply]
## PASO 3
Pone un BREAKPOINT sobre la primera línea de tu código fuente dentro de la función `run`
![image](https://user-images.githubusercontent.com/3951895/115033066-fc97f700-9e97-11eb-81ec-f789912fda7a.png)

**Reply by acedward** [reply]
## PASO 4 
Vuelve a introducir valores en tu formulario y debería detenerse ahí
Ahora puedes ver, paso por paso que valores tienen tus variables. 
Para saber que está pasando

**Reply by acedward** [reply]
## NOTA 
Como estas instancias son de poca duración. A veces es difícil lograr que se detenga.
Y ayuda MUCHO simplemente agregar `console.log('Nombre descriptivo:', variable );`

**Reply by NicolasPLG** [reply]
Gracias @acedward por la explicación, realizaré los pasos a seguir y te comento.

**Reply by NicolasPLG** [reply]
@acedward he tratado de ver en la consola pero no me arroja nada de lo que me indicas no encuentro el VM9 o me faltaría algo?

![image](https://user-images.githubusercontent.com/61419471/115040220-33254000-9e9f-11eb-9f1b-c1943f318c41.png)

**Reply by acedward** [reply]
Agregaste los console.log?

**Reply by NicolasPLG** [reply]
eso se hace desde la rutina vdd?, como el caso anterior del get_asset. no lo he agregado. Lo agregaré

**Reply by NicolasPLG** [reply]
@acedward una pregunta, como escribo la variable? con el identifier del valor que quiero obtener? 
Ejemplo: console.log('error del cálculo: ',var:ipa_tabaco_hepato)?

**Reply by acedward** [reply]
probablemente `console.log('error del cálculo:', ipa_tabaco_hepato);`

**Reply by NicolasPLG** [reply]
Me arroja esto

![image](https://user-images.githubusercontent.com/61419471/115248723-eedfad00-a0f5-11eb-8642-97149cddf122.png)
Creo que lo estoy haciendo netamente mal :(

**Reply by acedward** [reply]
Puedes pegar el código por favor.

**Reply by acedward** [reply]
Original:
```
const peso = Number(context['responses#peso_hepatopancreaticobiliar']);
const altura = Number(context['responses#talla_hepatopancreaticobiliar']);
var imc = peso / (altura**2);
commands.push( {         cmd: 'SET_RESPONSES',   value: imc  });  
commands.push(  {  cmd: 'SET_READONLY',    value: 'true' });  
return commands;
```
Con Logs:
```
const peso = Number(context['responses#peso_hepatopancreaticobiliar']);
console.log('peso:', peso);
const altura = Number(context['responses#talla_hepatopancreaticobiliar']);
cosole.log('altura:', altura);
var imc = peso / (altura**2);
console.log('imc:', imc);
commands.push( {         cmd: 'SET_RESPONSES',   value: imc  });  
commands.push(  {  cmd: 'SET_READONLY',    value: 'true' });  
console.log(commands);
return commands;
```

**Reply by NicolasPLG** [reply]
los valores me arroja bien, creo que es el problema de la impresion.

![image](https://user-images.githubusercontent.com/61419471/115258291-db850f80-a0fe-11eb-9a06-500e5087ed1e.png)

Debería colocar la variable responses#user?

**Reply by NicolasPLG** [reply]
o el commands no se encuentra dentro de las funciones del exec o está mal escrito?

**Reply by acedward** [reply]
en el código original tampoco estaba definido, deberíá ser así (agregué solo la linea al comienzo)
```
const commands  = [] ;
const peso = Number(context['responses#peso_hepatopancreaticobiliar']);
console.log('peso:', peso);
const altura = Number(context['responses#talla_hepatopancreaticobiliar']);
cosole.log('altura:', altura);
var imc = peso / (altura**2);
console.log('imc:', imc);
commands.push( {         cmd: 'SET_RESPONSES',   value: imc  });  
commands.push(  {  cmd: 'SET_READONLY',    value: 'true' });  
console.log(commands);
return commands;
```

**Reply by NicolasPLG** [reply]
Me arroja el resultado sin errores por la consola, pero no se me visualiza aún el valor. Otro factor que puede ser es que el campo en donde quiero obtener el resultado lo tengo con isReadOnly:true ?

![image](https://user-images.githubusercontent.com/61419471/115293620-8d373700-a125-11eb-9bd6-5b351d545406.png)

![image](https://user-images.githubusercontent.com/61419471/115293771-b8ba2180-a125-11eb-8780-f2e3568473d9.png)

Edit: las imágenes son de dos casos distintos

**Reply by acedward** [reply]
@NicolasPLG podríá ser un bug. Nunca hemos usado POSTSAVE para setear un valor después de mandar el formulario.

**Reply by acedward** [reply]
@guillevalin parece un bug :bug:

**Reply by NicolasPLG** [reply]
:O entiendo @acedward , lo levanto como un ticket? o esperemos a @guillevalin . Necesitaría una alternativa para llenar ese campo (Podría ser el mensaje, pero no creo que les gustará el cliente). Esto es lo último que me falta para cerrar una implementación :(.

Gracias y quedo atento a las acciones

**Reply by acedward** [reply]
@NicolasPLG ¿porque quieres que aparezca en este campo?

**Reply by NicolasPLG** [reply]
@acedward porque despues en reports, se puede bajar el formulario con todas las respuestas del formulario y sirve como ficha del cliente.

**Reply by acedward** [reply]
pero esto es POST SAVE ya se guardó el formulario

**Reply by acedward** [reply]
Entonces solo ser vería visualmente en el formulario localmente, pero no estaría en el reporte (no se guardaría)

**Reply by NicolasPLG** [reply]
@acedward pero entonces cómo sería la forma para mostrar el valor del calculo en el formulario. Que en el momento de enviar el formulario, me realice el calculo y coloque el valor e la celda. Sería tambien con un preload también?.

**Reply by acedward** [reply]
Por favor abre otro ticket para hacer otra pregunta!

**Reply by acedward** [reply]
Entoncess vamos a cerrar esto y marcar que hay un bug en POSTSAVE

**Reply by NicolasPLG** [reply]
Ok @acedward lo levantaré de forma interna y dejaré en referencia esta discusión.
Gracias y cierro aqui
